packages/itwin/grouping-mapping-widget/src/components/Properties/GroupProperties/GroupPropertyAction.tsx

```diff
@@ -217,36 +292,87 @@ export const GroupPropertyAction = ({ mappingId, group, groupProperty, onSaveSuc
             selection is made and saved.
           </Alert>
         )}
-        <Fieldset className="gmw-property-view-container" legend="Mapped Properties">
-          <div className="gmw-property-view-button">
-            <Button onClick={async () => setShowPropertiesSelectionModal(true)} disabled={isLoading}>
+        <ExpandableBlock
+          title={"Mapped Properties"}
+          endIcon={
+            <Icon fill={selectedProperties.length > 0 ? "informational" : "default"}>
+              <SvgLabel />
+            </Icon>
+          }
+          isExpanded={selectedProperties.length > 0 ? true : false}>
+          <InputGroup className='gmw-property-view-container'>
+            <div className="gmw-property-view-button">
+              <Button
+                onClick={async () => setShowPropertiesSelectionModal(true)}
+                disabled={isLoading}
+              >
               Select Properties
-            </Button>
-          </div>
-          <div className="gmw-properties-list">
-            {selectedProperties.length === 0 && !isLoading ? (
-              <div className="gmw-empty-selection">
-                <Text>No properties selected.</Text>
-                <Text>Press the &quot;Select Properties&quot; button for options.</Text>
-              </div>
-            ) : (
-              selectedProperties.map((property) => (
-                <GroupPropertyListItem
-                  key={property.key}
-                  content={`${property.displayLabel}`}
-                  title={`${property.actualECClassName}`}
-                  description={property.categoryLabel}
-                />
-              ))
-            )}
-          </div>
-        </Fieldset>
+              </Button>
+            </div>
+            <div className="gmw-properties-list">
+              {selectedProperties.length === 0 && !isLoading ?
+                <div className='gmw-empty-selection'>
+                  <Text>No properties selected.</Text>
+                  <Text>Press the &quot;Select Properties&quot; button for options.</Text>
+                </div> :
+                selectedProperties.map((property) => (
+                  <GroupPropertyListItem
+                    key={property.key}
+                    content={`${property.displayLabel}`}
+                    title={`${property.actualECClassName}`}
+                    description={property.categoryLabel}
+                  />
+                ))}
+            </div>
+          </InputGroup>
+        </ExpandableBlock>
+        <div ref={calculatedPropertyActionRef}>
+          <ExpandableBlock title={"Calculated Property"}
```

Comment:
This can be made into its own component. 
Something like:
```
interface ScrollableExpandableBlockProps extends ExpandableBlockProps {
  ref: React.RefObject<HTMLDivElement>; <-- parent ref
}

const ScrollableExpandableBlock = forwardRef<HTMLDivElement, ScrollableExpandableBlockProps>(
...
    const handleToggle = useCallback((isExpanding) => {
      if (isExpanding) {
        ...
      }
      onToggle(isExpanding);
    }, [...])

    return (
      <div ref={ref} <-- forwardRef>
        <ExpandableBlock
          title={title}
          endIcon={endIcon}
          isExpanded={isExpanded}
          onToggle={handleToggle}>
          {children}
        </ExpandableBlock>
      </div>
    );
```
https://react.dev/reference/react/forwardRef

Reply:
I got working with the fowardRef and created a new component, it could be refactor a bit more so I'll play around with it a little bit more.

Reply:
Added useFowardRef hook so that the forwarded ref can be use by the component.